File: Java/Problem_069.java
public class Euler69 {
    static boolean isPrime(int x) {
        if (x < 2) return false;
        if (x % 2 == 0) return x == 2;
        for (int i = 3; (long)i*i <= x; i += 2)
            if (x % i == 0) return false;
        return true;
    }

    static int nextPrime(int p) {
        int x = p + 1;
        while (!isPrime(x)) x++;
        return x;
    }

    public static void main(String[] args) {
        final int LIMIT = 1_000_000;
        long n = 1;
        int p = 2;

        while (n * (long)p <= LIMIT) {
            n *= p;
            p = nextPrime(p);
        }

        System.out.println(n); // 510510
    }
}

File: Python/Problem_069.py
def is_prime(x: int) -> bool:
    if x < 2:
        return False
    if x % 2 == 0:
        return x == 2
    i = 3
    while i * i <= x:
        if x % i == 0:
            return False
        i += 2
    return True

def next_prime(p: int) -> int:
    x = p + 1
    while not is_prime(x):
        x += 1
    return x

def euler_69(limit: int = 1_000_000) -> int:
    n = 1
    p = 2
    while n * p <= limit:
        n *= p
        p = next_prime(p)
    return n

if __name__ == "__main__":
    print(euler_69())  # 510510
